test/hooks.test.js

```diff
@@ -730,11 +739,11 @@ test('onRoute hook should able to change the route url', t => {
 
     sget({
       method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port + encodeURI('/föö')
+      url: getUrl(fastify) + encodeURI('/foo')
```

Comment:
Why this change?

Reply:
@delvedor > It's related to the `dns` changes introduced with Node.js 17

Reply:
the tests were failing with non-english character included router paths. if this is supported it should be handled in a different test since this is a test for changing the route.

Reply:
Regardless, we have lost test coverage as a result of this change. It will be okay to move it to another test, but that test should be included as part of the change.